### PR TITLE
FLS-827-hsra-configs

### PR DESCRIPTION
**Pull request by hamzabinkhalid**
https://mhclgdigital.atlassian.net/browse/FLS-827

### Change description
- HSRA VR and RP configured in Apply and Assess
- Old HSRA R1 removed
- tests updated
- hsra seed data added
- added seed_assessment_store script to add scoring systems in DB
- seed_assessment_store script documentation added

- [X] Unit tests and other appropriate tests added or updated
- [X] README and other documentation has been updated / added (if needed)
- [X] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
- Verify HSRA round VR and RP are available on apply and assess and are working as expected

### Screenshots of UI changes (if applicable)
image

image

image

### CODE REVIEW

File: assessment_store/tasks/db_tasks.py [hunk 1]
```diff
@@ -135,6 +133,7 @@ def seed_local_assessment_store_db(c):
             # Insert scoring systems
             one_to_five_id = str(uuid.uuid4())  # Generate a UUID for OneToFive
             zero_to_three_id = str(uuid.uuid4())  # Generate a UUID for OneToThree
+            zero_to_one_id = str(uuid.uuid4())  # Generate a UUID for ZeroToOne
```
Comment (albertkol) [inline assessment_store/tasks/db_tasks.py:136]: Are you using this `ZeroToOne` anywhere?

File: assessment_store/tasks/db_tasks.py [hunk 2]
```diff
@@ -149,6 +148,12 @@ def seed_local_assessment_store_db(c):
                     "minimum_score": 0,
                     "maximum_score": 3,
                 },
+                {
```
Comment (ksp37-dluhc) [inline assessment_store/tasks/db_tasks.py:151]: I'm not sure how this works, but it looks like the scoring systems are only added if they don't exist first? And at the moment it only checks for OneToFive and ZeroToThree ... so maybe there needs to be a check for ZeroToOne?

File: fund_store/config/fund_loader_config/FAB/__init__.py [hunk 3]
```diff
@@ -51,8 +51,10 @@
             raise ValueError(f"No round_config found in {file}")
         fund_short_name = loader_config["fund_config"]["short_name"]
         round_short_name = loader_config["round_config"]["short_name"]
-        FAB_FUND_ROUND_CONFIGS[fund_short_name] = loader_config["fund_config"]
-        if not FAB_FUND_ROUND_CONFIGS[fund_short_name].get("rounds", None):
+
+        # Ensure the fund exists in the main config
+        if fund_short_name not in FAB_FUND_ROUND_CONFIGS:
+            FAB_FUND_ROUND_CONFIGS[fund_short_name] = loader_config["fund_config"]
             FAB_FUND_ROUND_CONFIGS[fund_short_name]["rounds"] = {}
```
Comment (ksp37-dluhc) [inline fund_store/config/fund_loader_config/FAB/__init__.py:58]: Is this right? Won't it overwrite the rounds if they existed in the fund_config?

File: fund_store/scripts/amend_round_dates.py [hunk 4]
```diff
@@ -47,7 +28,6 @@
     "NSTF_R2": "fc7aa604-989e-4364-98a7-d1234271435a",
     "CYP_R1": "888aae3d-7e2c-4523-b9c1-95952b3d1644",
     "DPIF_R2": "0059aad4-5eb5-11ee-8c99-0242ac120002",
-    "HSRA_R1": "50062ff6-e696-474d-a560-4d9af784e6e5",
```
Comment (ksp37-dluhc) [inline fund_store/scripts/amend_round_dates.py:50]: Won't we need to add HSRA_VR and HSRA_RP?

File: assessment_store/tasks/db_tasks.py (hunk 1 above)
Comment (hamzabinkhalid) [reply assessment_store/tasks/db_tasks.py:136]: currently using the default 1-5 for these rounds until we get the confirmation from the onboarding team.

File: fund_store/config/fund_loader_config/FAB/__init__.py (hunk 3 above)
Comment (hamzabinkhalid) [reply fund_store/config/fund_loader_config/FAB/__init__.py:58]: This block ensures that the fund_short_name is added to FAB_FUND_ROUND_CONFIGS only if it does not already exist. If the fund already exists, this block is skipped.

File: assessment_store/tasks/db_tasks.py (hunk 2 above)
Comment (hamzabinkhalid) [reply assessment_store/tasks/db_tasks.py:151]: Good spot 👍, will add a check for ZeroToOne as well.

**Review by ksp37-dluhc — APPROVED** [review]

**Pull request merged**